src/main/java/io/strimzi/kafka/quotas/VolumeSourceBuilder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.quotas;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.LogDirDescription;
+
+/**
+ * A builder which ensures the <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-827%3A+Expose+log+dirs+total+and+usable+space+via+Kafka+API">KIP-827 API</a> is available and will throw exceptions if not.
+ *
+ * @see <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-827%3A+Expose+log+dirs+total+and+usable+space+via+Kafka+API">KIP-827 API</a>
+ */
+public class VolumeSourceBuilder implements AutoCloseable {
```

Comment:
The disconnect in naming came from an attempt at backwards compatibility since abandoned originally the build returned a `LocalVolumeSource` or the `ClusterVolumeSource` depending on the configuraiton. Will clean up the naming now.

Reply:
resolved with https://github.com/strimzi/kafka-quotas-plugin/pull/33/commits/7d4630fe4b90ebe8638a4a79d1109f7755ad0589